File: Leetcode/Daily/07-2025/Python3/14.py
# Problem: 1290. Convert Binary Number in a Linked List to Integer (Easy)
# Link: https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        val = 0
        
        while head is not None:
            val *= 2
            val += head.val
            head = head.next

        return val

File: Leetcode/Daily/07-2025/TypeScript/14.ts
// Problem: 1290. Convert Binary Number in a Linked List to Integer (Easy)
// Link: https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer

/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function getDecimalValue(head: ListNode | null): number {
    let val = 0;
    
    while (head != null) {
        val *= 2;
        val += head.val;
        head = head.next;
    }

    return val;     
};